controllers/dscinitialization/utils.go

```diff
@@ -74,13 +74,36 @@ func (r *DSCInitializationReconciler) createOdhNamespace(dscInit *dsci.DSCInitia
 			r.Log.Error(err, "Unable to fetch namespace", "name", name)
 			return err
 		}
-	} else if dscInit.Spec.Monitoring.Enabled {
+	} else if dscInit.Spec.Monitoring.Enabled && dscInit.Spec.Monitoring.Namespace == name {
 		err = r.Patch(ctx, foundNamespace, client.RawPatch(types.MergePatchType,
 			[]byte(`{"metadata": {"labels": {"openshift.io/cluster-monitoring": "true"}}}`)))
 		if err != nil {
 			return err
 		}
 	}
+	// Create Monitoring Namespace if it is enabled and not exists
+	if dscInit.Spec.Monitoring.Enabled {
+		foundMonitoringNamespace := &corev1.Namespace{}
+		monitoringName := dscInit.Spec.Monitoring.Namespace
+		err := r.Get(ctx, client.ObjectKey{Name: monitoringName}, foundMonitoringNamespace)
+		if err != nil { // enabled, but monitoring namespace
```

Comment:
Wouldn't be better to check if the error here IsNotFound() instead of trying to create the namespace and look for the IsAlreadyExists()?

Reply:
yeah correct :+1:

I will update it